ext/charlock_holmes/extconf.rb

```diff
@@ -49,4 +49,30 @@
 $CFLAGS << ' -Wall -funroll-loops'
 $CFLAGS << ' -Wextra -O0 -ggdb3' if ENV['DEBUG']
 
+minimal_program = <<~SRC
+  #include <unicode/translit.h>
+  int main() { return 0; }
+SRC
+
+# Pass -x c++ to force gcc to compile the test program
+# as C++ (as it will end in .c by default).
+compile_options = +"-x c++"
+
+icu_requires_version_flag = checking_for("icu that requires explicit C++ version flag") do
+  !try_compile(minimal_program, compile_options)
+end
+
+if icu_requires_version_flag
+  abort "Cannot compile icu with your compiler: recent versions require C++17 support." unless %w[c++20 c++17 c++11 c++0x].any? do |std|
```

Comment:
Hmmm, I thought this was working, but I'm seeing:

```
compiling ../../../../ext/charlock_holmes/converter.c
error: invalid argument '-std=c++20' not allowed with 'C'
```

Reply:
Hmm, now I'm seeing a separate problem:

```
/Users/stanhu/github/charlock_holmes/lib/charlock_holmes.rb:1:in `require': dlsym(0x8849a5f0, Init_charlock_holmes): symbol not found - Init_charlock_holmes (LoadError)
```

Seems like `Init_charlock_holmes` is optimized out.

Reply:
Seems like that's an existing problem on macOS. Seems to work fine as a gem.